### PR TITLE
Bring the user API up to date with newly added backend functionality

### PR DESCRIPTION
**Pull request by bp323**

### CODE REVIEW

**Comment by bp323** [comment]
Assigning to @Coenego for review.

File: shared/gh/api/gh.api.user.js [hunk 1]
```diff
@@ -384,6 +458,48 @@ define(['exports'], function(exports) {
     };
 
     /**
+     * Change a user's local password
+     *
+     * @param  {Number}      userId                     The ID of the user for which to change the local password
```
Comment (Coenego) [inline shared/gh/api/gh.api.user.js]: Indention issue

File: shared/gh/api/gh.api.user.js [hunk 2]
```diff
@@ -148,42 +151,93 @@ define(['exports'], function(exports) {
...
+            'error': function(jqXHR, textStatus) {
+                return callback({'code': jqXHR.status, 'msg': jqXHR.responseText});
+            }
+        });
+    };
+
+    /**
+     * Reset the user calendar token
+     *
+     * @param  {[type]}      userId               The ID of the user to reset the calendar token for
```
Comment (Coenego) [inline shared/gh/api/gh.api.user.js]: I assume this should be: `* @param  {Number}` ?

**Comment by Coenego** [comment]
Left some minor comments. Reassigning to @bp323

**Comment by bp323** [comment]
Followed up on review, back to @Coenego

**Pull request merged**